src/ch/ehi/ili2db/fromili/FromIliRecordConverter.java

```diff
@@ -194,7 +196,28 @@ public void generateTable(ViewableWrapper def,int pass)
 		if(base==null && !def.isSecondaryTable()){
 			if(createTypeDiscriminator || def.includesMultipleTypes()){
 				  dbCol=createSqlTypeCol(DbNames.T_TYPE_COL);
-				  dbTable.addColumn(dbCol);
+
+                  ArrayList<String> extensions = new ArrayList<String>();
+                  for (Object o : def.getViewable().getExtensions()){
+                      extensions.add(ili2sqlName.mapIliClassDef((Viewable) o));
+                  }
+
+                  // Add t_type possible values to meta-info table
+                  metaInfo.setColumnInfo(dbTable.getName().getName(),
+                                         DbNames.T_TYPE_COL,
+                                         DbExtMetaInfo.TAG_COL_TYPES,
+                                         extensions.toString());
+
+                  if(createTypeConstraint){
+
+                      // Add check constraint on t_type column
+                      // FIXME uncomment next lines when DbColVarchar.setValueRestriction(String[] possibleValues)
+                      // in ehisqlgen.jar is implemented
+
```

Comment:
Waiting to be uncommented when `DbColVarchar.setValueRestriction(String[] possibleValues)` is ready

Reply:
thanks